app/scripts/lib/nonce-tracker.js

```diff
@@ -28,18 +29,26 @@ class NonceTracker {
     // calculate next nonce
     // we need to make sure our base count
     // and pending count are from the same block
+    const localNonceHex = this._getLocalNonce(address)
+    let localNonce = parseInt(localNonceHex, 16)
+    try {
+      assert(Number.isInteger(localNonce), `nonce-tracker - localNonce is not an integer - got: (${typeof localNonce}) "${localNonce}"`)
+    } catch (e) {
+      // throw out localNonce if not a number
+      localNonce = 0
+    }
     const currentBlock = await this._getCurrentBlock()
     const pendingTransactions = this.getPendingTransactions(address)
     const pendingCount = pendingTransactions.length
     assert(Number.isInteger(pendingCount), `nonce-tracker - pendingCount is not an integer - got: (${typeof pendingCount}) "${pendingCount}"`)
     const baseCountHex = await this._getTxCount(address, currentBlock)
     const baseCount = parseInt(baseCountHex, 16)
     assert(Number.isInteger(baseCount), `nonce-tracker - baseCount is not an integer - got: (${typeof baseCount}) "${baseCount}"`)
-    const nextNonce = baseCount + pendingCount
+    const nextNonce = Math.max(baseCount, localNonce + 1) + pendingCount
```

Comment:
This is not really what we discussed before.
we suggested 2 separate strategies and then taking the higher result as the nonce

**not** using 2 strategies to determine the base nonce
if you think your approach is better or the same, let me know